pkg/querier/batch/merge.go

```diff
@@ -31,8 +31,8 @@ func newMergeIterator(cs []GenericChunk) *mergeIterator {
 	c := &mergeIterator{
 		its:        its,
 		h:          make(iteratorHeap, 0, len(its)),
-		batches:    make(batchStream, 0, len(its)*2*promchunk.BatchSize),
-		batchesBuf: make(batchStream, len(its)*2*promchunk.BatchSize),
+		batches:    make(batchStream, 0, len(its)),
+		batchesBuf: make(batchStream, len(its)),
```

Comment:
I don't recall exactly why the pre-allocation was so big - wondering if you know why?

I can't think of a reason why this would affect correctness either, and the perf results speak for themselves...

Reply:
From the correctness perspective, this change should be fine. `batchesBuf` looks to be written by `mergeStreams()` which extends the slice if required.